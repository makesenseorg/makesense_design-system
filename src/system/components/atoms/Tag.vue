<template>
  <div
    class="tag"
    :class="`tag--${color} tag--${size} tag--${isCliquable}`"
    @click="onClick"
  >
    {{ getLabel() }}
    <Icon v-if="icon" class="tag__icon" :type="icon" :color="color"></Icon>
  </div>
</template>

<script>
import Icon from "@@/components/atoms/icon";

export default {
  name: "Tag",
  components: { Icon },
  props: {
    color: {
      type: String,
      required: false,
      default: "default",
      validator: function(value) {
        return ["default", "main"].indexOf(value) !== -1;
      }
    },
    size: {
      type: String,
      required: false,
      default: "medium",
      validator: function(value) {
        return ["medium", "small", "large"].indexOf(value) !== -1;
      }
    },
    label: {
      type: String,
      required: true
    },
    trunc: {
      type: Boolean,
      default: true
    },
    icon: {
      type: [String, Boolean],
      required: false,
      default: false,
      validator: function(value) {
        return ["close", "check"].indexOf(value) !== -1;
      }
    },
    cliquable: {
      type: Boolean,
      required: false,
      default: false
    }
  },
  computed: {
    isCliquable() {
      return this.cliquable ? "cliquable" : "not-cliquable";
    }
  },
  methods: {
    onClick: function() {
      this.$emit("click");
    },

    getLabel: function() {
      if (!this.label) return "-No label-";
      if (!this.trunc) return this.label;

      if (this.label.length > 30) return this.label.substring(0, 30) + "...";
      else return this.label;
    }
  }
};
</script>

<style lang="scss" scoped>
.tag {
  @include border-round;
  @include text-small;
  display: inline-flex;
  align-items: center;
  font-weight: $font-weight-extrabold;
  padding: 0.625em;
  background: $color-neutral-80;
  color: $color-text;

  &--main {
    background: $color-secondary;
    color: $color-neutral-100;
  }

  &--small {
    @include text-xsmall;
  }

  &--cliquable {
    cursor: pointer;
  }

  &:hover {
    background: $color-secondary;
    color: $color-neutral-100;
  }
}

.tag__icon {
  margin-left: $space_xxs;
  color: inherit;
}
</style>
