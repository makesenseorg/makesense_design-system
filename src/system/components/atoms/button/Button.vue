<template>
  <component
    :is="tag"
    :class="[
      `button button--type-${type} button--size-${size}`,
      { 'button--loading': loading }
    ]"
    :role="role"
    tabindex="0"
    :type="inputType"
    :disabled="disabled"
    ref="container"
    :style="{
      minWidth: $slots.loading === undefined ? minWidth + 'px' : 'auto'
    }"
    @click="onClick"
    @keyup.enter="onClick"
  >
    <mks-icon
      v-if="icon && !iconRight"
      :type="icon"
      class="button__icon button__icon--left"
      :size="size === 'round' ? 24 : 20"
    ></mks-icon>
    <!-- @slot Content of the button-->
    <slot v-if="!loading" />
    <!-- @slot Content of the button when `loading` prop is true -->
    <slot v-else name="loading">
      <mks-loading
        type="standalone"
        :inline="true"
        color="#fff"
        :size="20"
      ></mks-loading>
    </slot>
    <mks-icon
      v-if="icon && iconRight"
      :type="icon"
      class="button__icon button__icon--right"
      :size="size === 'round' ? 24 : 20"
    ></mks-icon>
  </component>
</template>
<script>
/**
 * Buttons can be used for any action or inner link.
 * @version 0.3.0
 */
export default {
  name: "MksButton",
  release: "0.3.0",
  props: {
    /**
     * Buttons can be replaced by any of these tags : "div", "a", "input", "button", "span"
     */
    tag: {
      type: String,
      default: "button",
      validator: function(value) {
        return ["div", "a", "input", "button", "span"].indexOf(value) !== -1;
      }
    },
    /**
     * If loading is set to true, the button will display the loading slot or "Processing..."
     */
    loading: {
      type: Boolean,
      default: false
    },
    /**
     * If disabled is set to true, the button will be disabled
     */
    disabled: {
      type: Boolean,
      default: false
    },
    /**
     * Can set the input type, in case of button tag (ex: "submit")
     */
    inputType: {
      type: String,
      required: false,
      default: ""
    },
    /**
     * Color of the button. "primary",
            "secondary",
            "tertiary",
            "neutral",
            "positive",
            "warning",
            "negative"
     */
    type: {
      type: String,
      default: "primary",
      validator: function(value) {
        return (
          [
            "primary",
            "secondary",
            "tertiary",
            "neutral",
            "positive",
            "warning",
            "negative",
            "text",
            "text-active"
          ].indexOf(value) !== -1
        );
      }
    },
    /**
     * Has to be a valid icon type
     */
    icon: {
      type: String
    },
    /**
     * Place the icon on the right side of the text
     */
    iconRight: {
      type: Boolean,
      default: false
    },
    /**
     * Size of the button. "default", "small", "medium" "full", "round"
     */
    size: {
      type: String,
      default: "default",
      validator: function(value) {
        return (
          ["default", "small", "medium", "full", "round", "square"].indexOf(
            value
          ) !== -1
        );
      }
    },
    /**
     * Size of the button. "default", "small", "full", "round"
     */
    role: {
      type: String,
      default: "button"
    }
  },
  data: function() {
    return {
      minWidth: 0,
      observer: null
    };
  },
  mounted() {
    this.updateMinWidth();

    this.observer = new MutationObserver(() => {
      this.updateMinWidth();
    });

    this.observer.observe(this.$refs.container, {
      attributes: true,
      childList: true,
      characterData: true,
      subtree: true
    });
  },
  beforeDestroy: function() {
    // Clean up
    this.observer.disconnect();
  },
  methods: {
    onClick() {
      /**
       * Fires after click.
       *
       * @event click
       */
      this.$emit("click");
    },
    updateMinWidth() {
      this.$nextTick(function() {
        var container = this.$refs.container;
        if (container !== undefined) this.minWidth = container.offsetWidth;
      });
    }
  }
};
</script>

<style lang="scss" scoped>
.button {
  @include cursor-pointer;
  @include border-radius;
  @include button-reset;
  @include padding-button;
  @include text-body;
  display: inline-block;
  font-weight: $font-weight-extrabold; // override text-body
  border-width: $border-width-m;
  border-style: solid;
  text-decoration: none !important;
  transition: background-color 0.2s, border-color 0.2s;
  text-align: center;

  &:focus {
    @include focus($background-color-primary);
  }

  &--size-small {
    @include border-round;
    @include padding-button-small;
  }

  &--size-medium {
    @include border-radius;
    @include padding-button-small;
  }
  &--size-full {
    width: 100%;
    text-align: center;
  }

  &--size-round {
    @include title-small;
    display: inline-flex;
    align-items: center;
    justify-content: center;
    text-align: center;
    user-select: none;
    border-radius: $border-radius-circle;
    padding-right: 0;
    padding-left: 0;
    width: $space-xxl;
    height: $space-xxl;
    text-align: center;

    .button__icon {
      margin: 0;
    }
  }
  &--size-square {
    @include text-body;
    display: inline-flex;
    align-items: center;
    justify-content: center;
    text-align: center;
    user-select: none;
    border-radius: $border-radius-m;
    padding-right: 0;
    padding-left: 0;
    width: $space-xl;
    height: $space-xl;
    padding: $space-xxs;
    text-align: center;
  }

  @mixin type-text {
    @include text-small;
    background-color: transparent;
    text-transform: uppercase;
    color: inherit;
    padding: 0;
    margin: 0;
    border: none;
  }

  &--type-text {
    @include type-text;
    font-weight: $font-weight-normal;
  }

  &--type-text-active {
    @include type-text;
    font-weight: $font-weight-black;
  }

  &--type-primary {
    background-color: $color-primary;
    border-color: $color-primary;
    color: $color-primary-inverse;

    &:hover {
      background-color: $color-primary-active;
      border-color: $color-primary-active;
    }
  }

  &--type-secondary {
    background-color: $color-secondary;
    border-color: $color-secondary;
    color: $color-secondary-inverse;

    &:hover {
      // todo : garder la border peut-être? ou rendre l'effet un peu plus légeer
      background-color: $color-secondary-active;
      border-color: $color-secondary-active;
    }

    &:focus {
      @include focus($color-secondary);
    }
  }

  &--type-tertiary {
    background-color: $color-tertiary;
    border-color: $color-tertiary;
    color: $color-tertiary-inverse;

    &:hover {
      // todo : garder la border peut-être? ou rendre l'effet un peu plus légeer
      background-color: $color-tertiary-active;
      border-color: $color-tertiary-active;
    }

    &:focus {
      @include focus($color-tertiary);
    }
  }

  &--type-positive {
    background-color: $color-success;
    border-color: $color-success;
    color: $color-success-inverse;

    &:hover {
      background-color: $color-success-active;
      border-color: $color-success-active;
    }

    &:focus {
      @include focus($color-success);
    }
  }

  &--type-warning {
    background-color: $color-warning;
    border-color: $color-warning;
    color: $color-warning-inverse;

    &:hover {
      background-color: $color-warning-active;
      border-color: $color-warning-active;
    }

    &:focus {
      @include focus($color-warning-active);
    }
  }

  &--type-negative {
    background-color: $color-danger;
    border-color: $color-danger;
    color: $color-danger-inverse;

    &:hover {
      background-color: $color-danger-active;
      border-color: $color-danger-active;
    }

    &:focus {
      @include focus($color-danger-active);
    }
  }

  &--type-neutral {
    color: $color-neutral-20;
    background-color: $background-color-base;
    border-color: $background-color-softest;

    &:hover {
      background-color: $background-color-softest;
    }

    &:focus {
      @include focus($background-color-softest);
    }
  }

  &:disabled,
  &[disabled] {
    opacity: 0.5;
    cursor: default;
    pointer-events: none;
  }

  &--loading {
    opacity: 0.7;
  }
}

.button__icon {
  &--left {
    margin-right: $space-xs;
  }
  &--right {
    margin-left: $space-xs;
  }
}

button {
  outline: none !important;
}
</style>
<docs>
## Colors
  ```jsx
    <mks-button tag="button">Button default</mks-button>
    <mks-button tag="button" type="secondary">Button secondary</mks-button>
    <mks-button tag="button" type="tertiary">Button tertiary</mks-button>
    <mks-button tag="button" type="neutral">Button neutral</mks-button>
    <br>
    <mks-button type="positive">Button positive</mks-button>
    <mks-button type="warning">Button warning</mks-button>
    <mks-button type="negative">Button negative</mks-button>
  ```

## Sizes
```jsx
    <mks-button size="round" icon="menu"></mks-button>
    <mks-button size="small">Button small</mks-button>
    <mks-button size="medium">Button medium</mks-button>
    <mks-button size="default">Button default</mks-button>
    <mks-button size="full">Button full</mks-button>
  ```

## With icons
```jsx
    <mks-button size="round" icon="bell"></mks-button>
    <br>
    <mks-button icon="menu" size="small">Menu</mks-button>
    <mks-button icon="menu" icon-right size="small">Menu</mks-button>
    <br>
    <mks-button icon="star">Sauvegarder</mks-button>
    <mks-button icon="star" icon-right>Sauvegarder</mks-button>
  ```

  ## States
```jsx
    <mks-button v-bind:disabled="true">Disabled</mks-button>
    <br>
    <mks-button v-bind:loading="true">
        Load 
        <template v-slot:loading>Loading...</template>
    </mks-button>
    <br>
    <mks-button v-bind:loading="true">
        Load 
    </mks-button>
    <br>
    <mks-button v-bind:loading="true" size="small">
        Load 
    </mks-button>
  ```
</docs>
