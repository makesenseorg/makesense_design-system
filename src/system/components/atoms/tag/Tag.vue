<template>
  <div
    class="tag"
    :class="
      `tag--color-${color} tag--size-${size} tag--${isCliquable} tag--style-${variant}`
    "
    :role="isCliquable ? 'button' : undefined"
    :style="
      `--color: var(--color-${color}); 
      --color-hover: ${hoverColor}; 
      --color-inverse: var(--color-${contrastColor})`
    "
    @click="onClick"
  >
    <mks-icon
      v-if="icon && iconLeft"
      class="tag__icon tag__icon--left"
      :type="icon"
      :alt="iconAlt"
      :size="10"
    ></mks-icon>
    <slot>{{ formattedLabel }}</slot>
    <mks-icon
      v-if="icon && !iconLeft"
      class="tag__icon tag__icon--right"
      :type="icon"
      :alt="iconAlt"
      :size="10"
    ></mks-icon>
  </div>
</template>

<script>
import Vue from "vue";
/**
 * Tags are used for small informations or taxonomies.
 * @version 0.1.0
 */
export default {
  name: "MksTag",
  props: {
    /**
     * Any of the design tokens colors and these : 
            "neutral",
            "positive",
            "warning",
            "negative"
     */
    color: {
      type: String,
      required: false,
      default: "neutral",
      validator: function(value) {
        return (
          Vue.prototype.$colorExists(value) ||
          ["positive", "negative", "warning", "neutral"].indexOf(value) !== -1
        );
      }
    },
    /**
     * Size of the tag : "small" or "medium"
     */
    size: {
      type: String,
      required: false,
      default: "medium",
      validator: function(value) {
        return ["medium", "small"].indexOf(value) !== -1;
      }
    },
    /**
     * Style of the tag : full(default), ghost, bare
     */
    variant: {
      type: String,
      default: "full",
      validator: function(value) {
        return ["full", "ghost", "bare"].indexOf(value) !== -1;
      }
    },
    /**
     * Text of the tag
     */
    label: {
      type: String,
      required: false
    },
    /**
     * Should the text be truncated if too long
     */
    trunc: {
      type: Boolean,
      default: true
    },
    /**
     * Icon displayed on the right of the tag : "close" or "check"
     */
    icon: {
      type: String,
      required: false
    },
    /**
     * Icon description
     */
    iconAlt: {
      type: String,
      required: false
    },
    /**
     * Display icon on the left side of the Tag
     */
    iconLeft: {
      type: Boolean,
      required: false
    },
    /**
     * Should the tag be cliquable
     */
    cliquable: {
      type: Boolean,
      required: false,
      default: false
    }
  },
  computed: {
    isCliquable() {
      return this.cliquable ? "cliquable" : "not-cliquable";
    },
    formattedLabel() {
      if (!this.label) return "-No label-";
      if (!this.trunc) return this.label;

      if (this.label.length > 30) return this.label.substring(0, 30) + "...";
      else return this.label;
    },
    contrastColor() {
      return this.$getContrastColor(this.color);
    },
    hoverColor() {
      return this.$getHoverColor(this.color);
    }
  },
  methods: {
    onClick: function() {
      this.$emit("click");
    }
  }
};
</script>

<style lang="scss" scoped>
.tag {
  @include border-round;
  @include text-small;
  line-height: 1;
  display: inline-flex;
  align-items: center;
  font-weight: $font-weight-extrabold;
  padding: 0.35em 0.625em;
  background-color: var(--color);
  border-color: var(--color);
  color: var(--color-inverse);
  border: 2px solid var(--color);
  transition: all 0.2s ease;

  &--cliquable {
    cursor: pointer;
    &:hover {
      background-color: var(--color-hover);
      border-color: var(--color-hover);
    }
  }

  &--color-neutral {
    --color: var(--color-neutral-80) !important;
    --color-inverse: var(--color-neutral-20) !important;
    --color-hover: var(--color-neutral-70) !important;

    &.tag--style-ghost {
      background-color: transparent;
      border-color: var(--color-neutral-80);
      color: var(--color-inverse);
      &:hover {
        background-color: var(--color);
      }
    }
  }

  &--style-full {
    background-color: var(--color);
    border-color: var(--color);
    color: var(--color-inverse);
  }
  &--style-ghost {
    background-color: transparent;
    border-color: var(--color);
    color: var(--color);
  }
  &--style-bare {
    background-color: transparent;
    border-color: transparent;
    color: var(--color-inverse);
    padding: 0.35em 0.15em;
    &:hover {
      background: transparent;
      border-color: transparent;
      text-decoration: underline;
    }
  }

  &--color-positive {
    --color: var(--color-success) !important;
    --color-inverse: var(--color-success-inverse) !important;
    --color-hover: var(--color-success-active) !important;
  }

  &--color-warning {
    --color: var(--color-warning) !important;
    --color-inverse: var(--color-warning-inverse) !important;
    --color-hover: var(--color-warning-active) !important;
  }

  &--color-negative {
    --color: var(--color-danger) !important;
    --color-inverse: var(--color-danger-inverse) !important;
    --color-hover: var(--color-danger-active) !important;
  }

  &--size-small {
    @include text-xsmall;
    line-height: 1;
  }

  &--not-cliquable {
    pointer-events: none;
  }
}

.tag__icon {
  color: inherit;
  transform: scale(1.5);
  &--left {
    margin-right: $space_xxs;
  }
  &--right {
    margin-left: $space_xxs;
  }
}
</style>
<docs>
## Variants
```jsx
  <mks-tag label="Full" variant="full" v-bind:cliquable="true"></mks-tag>
  <mks-tag label="Ghost" variant="ghost" color="neutral" v-bind:cliquable="true"></mks-tag>
  <mks-tag label="Bare" variant="bare" v-bind:cliquable="true"></mks-tag>
```

## Colors
  ```jsx
    <mks-tag label="Neutral (default)"></mks-tag>

    <mks-text tag="p">Theme colors</mks-text>
    <mks-tag label="Primary" color="primary" v-bind:cliquable="true"></mks-tag>
    <mks-tag label="Secondary" color="secondary" v-bind:cliquable="true"></mks-tag>
    <mks-tag label="Tertiary" color="tertiary" v-bind:cliquable="true"></mks-tag><br><br>

    <mks-text tag="p">Feedback colors</mks-text>
    <mks-tag label="Positive" color="positive" v-bind:cliquable="true"></mks-tag>
    <mks-tag label="Warning" color="warning" v-bind:cliquable="true"></mks-tag>
    <mks-tag label="Negative" color="negative" v-bind:cliquable="true"></mks-tag><br><br>

    <mks-text tag="p">Other brand colors<br>
      <mks-text tag="span" size="small" color="light">All the colors in design tokens are available.</mks-text>
    </mks-text>
    
    <mks-tag label="Neutral 20" color="neutral-20" v-bind:cliquable="true"></mks-tag>
    <mks-tag label="Brick red" color="brick-red" v-bind:cliquable="true"></mks-tag>
    <mks-tag label="Atlantis" color="atlantis" v-bind:cliquable="true"></mks-tag>
    <mks-tag label="Sahara sand" color="sahara-sand" v-bind:cliquable="true"></mks-tag><br><br>
    
  ```

## Sizes
```jsx
    <mks-tag label="Small" size="small"></mks-tag>
    <mks-tag label="Medium (default)" size="medium"></mks-tag>
  ```

## With icons
```jsx
    <mks-tag label="No icon" v-bind:cliquable="true"></mks-tag>
    <mks-tag label="Close icon" icon="x" v-bind:cliquable="true"></mks-tag>
    <mks-tag label="Check icon" icon="check" v-bind:cliquable="true"></mks-tag>
    <br>
    <mks-tag label="No icon" icon-left v-bind:cliquable="true" size="small"></mks-tag>
    <mks-tag label="Filter icon" icon-left icon="filter" v-bind:cliquable="true" size="small"></mks-tag>
    <mks-tag label="Search icon" icon-left icon="search" v-bind:cliquable="true" size="small"></mks-tag>
  ```

## Truncated
```jsx
    <mks-tag label="A super long label that needs to be truncated" v-bind:trunc="true"></mks-tag>
  ```
</docs>
