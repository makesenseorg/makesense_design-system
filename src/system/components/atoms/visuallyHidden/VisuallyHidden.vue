<template>
  <div class="visually-hidden">
    <!-- @slot hides anything present in the slot from regular devices, and shows it to assistive technologies -->
    <slot></slot>
  </div>
</template>

<script>
/** Visually Hidden is used when an element needs to be available to assistive technology, but otherwise hidden.
The visually hidden component hides text visually from screen, but keeps it available to assistive technologies, such as screen readers. The component shouldnâ€™t be used to hide interactive content. */
export default {
  name: "MksVisuallyHidden",
  release: "1.1.3"
};
</script>

<style lang="scss" scoped>
.visually-hidden {
  clip: rect(0 0 0 0);
  clip-path: inset(50%);
  height: 1px;
  overflow: hidden;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}
</style>

<docs>
## Provide more context

Inspect element to see additionnal text that will be read by screen savers.

```jsx
<mks-link to="/atoms/mkslink">
  Read more
  <mks-visually-hidden>about penguins in antartica</mks-visually-hidden>
</mks-link>
```

## The opposite of aria-hidden

<code>aria-hidden</code> indicates that an element should be ignored by assistive technologies. 
We can use it in combination with <code>mks-visually-hidden</code> to replace with more suited content.

```jsx
<mks-icon type="search" aria-hidden></mks-icon>
<mks-visually-hidden>Search</mks-visually-hidden>
```
</docs>
