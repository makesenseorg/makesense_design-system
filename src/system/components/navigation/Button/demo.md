## Button types

Use a primary button to draw the users attention to important actions. Use default buttons for less important actions.

A danger button should be used only for destructive actions.

```
<ds-button>Default</ds-button>
<ds-button primary>Primary</ds-button>
<ds-button secondary>Secondary</ds-button>
<ds-button danger>Danger</ds-button>
```

## Ghost buttons

Use a ghost button for secondary actions.

```
<ds-button ghost>Default</ds-button>
<ds-button ghost primary>Primary</ds-button>
<ds-button ghost secondary>Secondary</ds-button>
<ds-button ghost danger>Danger</ds-button>
```

## Button sizes

Use different sizes to create hierarchy.

```
<ds-button size="small">Small</ds-button>
<ds-button>Base</ds-button>
<ds-button size="large">Large</ds-button>
```

## Fullwidth button

Use a fullwidth button for important actions.

```
<ds-button primary fullwidth>Fullwidth button</ds-button>
```

## Button states

A button can take different states.

```
<ds-button>Default state</ds-button>
<ds-button disabled>Disabled state</ds-button>
<ds-button hover>Hover state</ds-button>
```

## Icon buttons

Add an icon to a button to help the user identify the button's action.

```
<ds-button icon="plus" primary>Click me</ds-button>
<ds-button icon-right="plus">Click me</ds-button>
<ds-button icon="plus" primary></ds-button>
<ds-button icon="plus" ghost></ds-button>
```

## Button as links

Provide a path to the button. You can pass a url string or a Vue router path object.

```
<ds-button path="/navigation">Click me</ds-button>
<ds-button :path="{ name: 'Navigation' }">Click me</ds-button>
```