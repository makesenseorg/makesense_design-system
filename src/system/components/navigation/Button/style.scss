.ds-button {
  @include reset;
  position: relative;
  width: auto;
  overflow: visible;
  -webkit-font-smoothing: inherit;
  -moz-osx-font-smoothing: inherit;
  -webkit-appearance: none;
  border: 0;
  cursor: pointer;
  user-select: none;
  font-size: $font-size-base;
  line-height: $line-height-base;
  font-family: $font-family-text;
  font-weight: $font-weight-bold;
  letter-spacing: $letter-spacing-large;
  display: inline-flex;
  vertical-align: middle;
  align-items: center;
  justify-content: center;
  text-decoration: none;
  padding: 0 $space-small;
  height: $input-height;
  border-radius: $border-radius-base;
  box-shadow: $box-shadow-small-inset, $box-shadow-x-small;
  border: $input-border-size solid transparent;
  transition: color $duration-short $ease-out,
    background-color $duration-short $ease-out;

  &:before {
    position: absolute;
    content: '';
    top: $space-xxx-small;
    left: $space-xxx-small;
    right: $space-xxx-small;
    bottom: $space-xxx-small;
    border-radius: $border-radius-base;
    box-shadow: $box-shadow-inset;
    opacity: 0;
    visiblity: hidden;
    pointer-events: none;
  }

  &:after {
    position: absolute;
    content: '';
    top: $space-xxx-small;
    left: $space-xxx-small;
    right: $space-xxx-small;
    bottom: $space-xxx-small;
    border-radius: $border-radius-base;
    border: 1px dotted currentColor;
    opacity: 0;
    visiblity: hidden;
    pointer-events: none;
    transition: all $duration-short $ease-out;
  }

  &:focus {
    outline: none;
    &:after {
      opacity: 0.8;
      visibility: visible;
    }
  }

  &:active {
    &:before {
      opacity: 0.6;
    }
  }

  &::-moz-focus-inner {
    border: 0;
    padding: 0;
  }

  &:disabled {
    opacity: 0.5;
    cursor: default;
    pointer-events: none;
  }

  // Default colors
  color: $text-color-base;
  background-color: $background-color-softer;

  &:hover,
  &.ds-button-hover {
    color: $text-color-base;
    background-color: $background-color-softer-active;
  }
}

.ds-button-fullwidth {
  width: 100%;
}

.ds-button-primary {
  color: $text-color-primary-inverse;
  background-color: $background-color-primary;

  &:hover,
  &.ds-button-hover {
    color: $text-color-primary-inverse;
    background-color: $background-color-primary-active;
  }

  &:active {
    &:before {
      opacity: 1;
    }
  }
}

.ds-button-secondary {
  color: $text-color-secondary-inverse;
  background-color: $background-color-secondary;

  &:hover,
  &.ds-button-hover {
    color: $text-color-secondary-inverse;
    background-color: $background-color-secondary-active;
  }

  &:active {
    &:before {
      opacity: 1;
    }
  }
}

.ds-button-danger {
  color: $text-color-danger-inverse;
  background-color: $background-color-danger;

  &:hover,
  &.ds-button-hover {
    color: $text-color-danger-inverse;
    background-color: $background-color-danger-active;
  }

  &:active {
    &:before {
      opacity: 1;
    }
  }
}

.ds-button-ghost {
  color: $text-color-base;
  background-color: transparent;
  box-shadow: none;

  &:focus {
    box-shadow: none;
  }

  &:hover,
  &.ds-hover {
    color: $text-color-base;
    background-color: $background-color-soft;
  }

  &:active {
    &:before {
      opacity: 0.6;
    }
  }

  &.ds-button-primary {
    color: $text-color-primary;
  }

  &.ds-button-secondary {
    color: $text-color-secondary;
  }

  &.ds-button-danger {
    color: $text-color-danger;
  }
}

.ds-button-size-small {
  font-size: $font-size-small;
  padding: 0 $space-x-small;
  height: $input-height-small;
}

.ds-button-size-large {
  font-size: $font-size-large;
  padding: 0 $space-base;
  height: $input-height-large;
}

.ds-button-icon-only {
  width: $input-height;
  padding: 0;
  border-radius: $border-radius-rounded;

  &:before, &:after {
    border-radius: $border-radius-rounded;
  }

  &.ds-button-size-small {
    width: $input-height-small;
  }

  &.ds-button-size-large {
    width: $input-height-large;
  }
}

.ds-button-text {
  line-height: inherit;
  display: inline-block;
  white-space: nowrap;
  margin: 0 $font-space-small;

  &:first-child {
    margin-left: 0;
  }

  &:last-child {
    margin-right: 0;
  }
}