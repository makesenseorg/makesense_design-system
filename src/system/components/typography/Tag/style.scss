.ds-tag {
	@include reset;
	@include stack-space($space-xxs);
	@include inline-space($font-space-xxs);
	display: inline-block;
	font-family: $font-text;
	line-height: $line-height-m;
	padding: 0 $font-space-m;
	border-radius: $border-radius-s;
	font-weight: $font-weight-bold;
	letter-spacing: $letter-spacing-large;
	text-transform: uppercase;
	color: $color-text;
	background-color: $background-color-softest;
}

.ds-tag-inverse {
	color: $text-color-inverse;
	background-color: $background-color-inverse-softer;
}

.ds-tag-primary {
	color: $text-color-primary-inverse;
	background-color: $background-color-primary;
}

.ds-tag-success {
	color: $text-color-success-inverse;
	background-color: $background-color-success;
}

.ds-tag-warning {
	color: $text-color-warning-inverse;
	background-color: $background-color-warning;
}

.ds-tag-danger {
	color: $text-color-danger-inverse;
	background-color: $background-color-danger;
}

.ds-tag-round {
	border-radius: $border-radius-m;
}

.ds-tag-size-base {
	font-size: $font-size-xs;
}

.ds-tag-size-small {
	font-size: $font-size-xs;
}

.ds-tag-size-large {
	font-size: $font-size-s;
}
