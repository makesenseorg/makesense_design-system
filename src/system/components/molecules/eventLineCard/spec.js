import { shallowMount, createLocalVue } from "@vue/test-utils";
import Comp from "./EventLineCard.vue";

import Button from "../../atoms/button/Button.vue";
import Tag from "../../atoms/tag/Tag.vue";
import VisuallyHidden from "../../atoms/visuallyHidden/VisuallyHidden.vue";

const $MKSlocale = { 
  event: 'Évènement',
  pastEvent: 'Évènement terminé',
};
const localVue = createLocalVue();
localVue.component("MksButton", Button);
localVue.component("MksTag", Tag);
localVue.component("MksVisuallyHidden", VisuallyHidden);

describe("EventLineCard.vue", () => {
  it("matches snapshot", () => {
    const wrapper = shallowMount(Comp, {
      localVue,
      mocks: { $MKSlocale },
      data() {
        return {
          uniqueId: 'id-override'
        }
      },
      propsData: {
        link: "https://example.com/",
        date: "26/01/1986",
        duration: 30,
        title: "The title",
        metas: ["tag1", "tag2"],
        cover: "https://placeholder.com/",
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("matches snapshot past version", () => {
    const wrapper = shallowMount(Comp, {
      localVue,
      mocks: { $MKSlocale },
      data() {
        return {
          uniqueId: 'id-override'
        }
      },
      propsData: {
        link: "https://example.com/",
        date: new Date(new Date("26/01/1986").getTime()).toString(),
        duration: 60,
        title: "The title",
        cover: "https://placeholder.com/",
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("emits click event", async () => {
    const wrapper = shallowMount(Comp, {
      localVue,
      mocks: { $MKSlocale },
      propsData: {
        link: "https://example.com/",
        date: "26/01/1986",
        duration: 30,
        title: "The title",
        cover: "https://placeholder.com/",
      },
    });

    wrapper.trigger("click");
    await wrapper.vm.$nextTick();

    expect(wrapper.emitted().click).toBeTruthy();
  });

  it("displays now modifier", () => {
    const wrapper = shallowMount(Comp, {
      localVue,
      mocks: { $MKSlocale },
      propsData: {
        link: "https://example.com/",
        date: new Date(Date.now() - 1200).toString(),
        duration: 60,
        title: "The title",
        cover: "https://placeholder.com/",
      },
    });

    expect(wrapper.classes()).toContain("event--now");
  });

  it("displays past modifier", () => {
    const wrapper = shallowMount(Comp, {
      localVue,
      mocks: { $MKSlocale },
      propsData: {
        link: "https://example.com/",
        date: new Date(Date.now() - 10000000).toString(),
        duration: 60,
        title: "The title",
        cover: "https://placeholder.com/",
      },
    });

    expect(wrapper.classes()).toContain("event--past");
  });

  it("displays button label content", () => {
    const wrapper = shallowMount(Comp, {
      localVue,
      mocks: { $MKSlocale },
      propsData: {
        link: "https://example.com/",
        date: new Date(Date.now() - 1200).toString(),
        duration: 60,
        callToAction: "abc",
        title: "The title",
        cover: "https://placeholder.com/",
      },
    });

    const button = wrapper.findComponent(Button);
    expect(button.exists()).toBe(true);
    expect(button.text()).toContain("abc");
  });
});
