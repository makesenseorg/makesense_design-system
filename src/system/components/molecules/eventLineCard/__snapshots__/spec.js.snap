// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventLineCard.vue matches snapshot 1`] = `
<div
  aria-labelledby="event-id-override"
  class="event"
  target="_blank"
>
  <div
    class="event__hour"
  >
    <time
      datetime="Invalid Date"
    >
      Invalid Date
    </time>
  </div>
   
  <div
    class="event__cover"
  >
    <img
      class="event__cover-image"
      src="https://placeholder.com/"
    />
  </div>
   
  <div
    class="event__content"
  >
    <div
      class="content__title"
    >
      <h3
        id="event-id-override"
      >
        <!---->
        
        The title
      
      </h3>
       
      <div
        class="event__metas"
      >
        <mks-tag-stub
          class="event__meta"
          color="neutral"
          label="tag1"
          size="medium"
          trunc="true"
        />
        <mks-tag-stub
          class="event__meta"
          color="neutral"
          label="tag2"
          size="medium"
          trunc="true"
        />
      </div>
    </div>
     
    <div
      class="content__action"
    >
      <mks-button-stub
        href="https://example.com/"
        inputtype=""
        role="button"
        size="small"
        tag="a"
        target="_blank"
        type="primary"
      >
        <mks-visually-hidden>
          Évènement: The title
        </mks-visually-hidden>
        
        Participer
      
      </mks-button-stub>
    </div>
  </div>
</div>
`;

exports[`EventLineCard.vue matches snapshot past version 1`] = `
<div
  aria-labelledby="event-id-override"
  class="event"
  target="_blank"
>
  <div
    class="event__hour"
  >
    <time
      datetime="Invalid Date"
    >
      Invalid Date
    </time>
  </div>
   
  <div
    class="event__cover"
  >
    <img
      class="event__cover-image"
      src="https://placeholder.com/"
    />
  </div>
   
  <div
    class="event__content"
  >
    <div
      class="content__title"
    >
      <h3
        id="event-id-override"
      >
        <!---->
        
        The title
      
      </h3>
       
      <!---->
    </div>
     
    <div
      class="content__action"
    >
      <mks-button-stub
        href="https://example.com/"
        inputtype=""
        role="button"
        size="small"
        tag="a"
        target="_blank"
        type="primary"
      >
        <mks-visually-hidden>
          Évènement: The title
        </mks-visually-hidden>
        
        Participer
      
      </mks-button-stub>
    </div>
  </div>
</div>
`;
